lang/cs/Org.Apache.REEF.Tests/Functional/Bridge/TestCloseTask.cs

```diff
@@ -322,6 +386,85 @@ public void OnError(Exception error)
             }
         }
 
+        private sealed class BreakTestTask : ITask, IObserver<ICloseEvent>
+        {
+            private volatile bool _break;
+            private volatile bool _stoped;
+            private readonly bool _enforceClose;
+            private readonly int _enforceCloseTimeout;
+
+            [Inject]
+            private BreakTestTask(
+                [Parameter(typeof(EnforceCloseTimeout))] int enforceCloseTimeout,
+                [Parameter(typeof(EnforceClose))] bool enforceClose)
+            {
+                _enforceClose = enforceClose;
+                _enforceCloseTimeout = enforceCloseTimeout;
+            }
+
+            public byte[] Call(byte[] memento)
+            {
+                int iterate = 1;
+
+                while (!_break && iterate < 100)
+                {
+                    iterate++;
+                    if (_enforceClose)
+                    {
+                        Thread.Sleep(5000);
+                    }
+                    else
+                    {
+                        Thread.Sleep(500);
+                    }
+                }
+
+                if (_break)
+                {
+                    Logger.Log(Level.Info, BreakTaskMessage);
+                }
+                _stoped = true;
+                return null;
+            }
+
+            public void Dispose()
+            {
+                Logger.Log(Level.Info, "Task is disposed.");
+            }
+
+            public void OnNext(ICloseEvent closeEvent)
+            {
+                if (closeEvent.Value.IsPresent() && Encoding.UTF8.GetString(closeEvent.Value.Value).Equals(DisposeMessageFromDriver))
+                {
+                    Logger.Log(Level.Info, "Closed event received in task:" + Encoding.UTF8.GetString(closeEvent.Value.Value));
+                    _break = true;
+                    if (_enforceClose)
+                    {
+                        Thread.Sleep(_enforceCloseTimeout);
```

Comment:
Yes.

The idea case is to inform Call() to finish the current iteration then return. However, in group communication case, when one of the task fails, all the other tasks are hung in a blocking call. We need a back up plan to enforce the closing. 

This issue has been discussed in Jira.